packages/server/assets/workspacesCore/typedefs/workspaces.graphql

```diff
@@ -46,13 +49,9 @@ type WorkspaceMutations {
     @hasScope(scope: "workspace:create")
   delete(workspaceId: String!): Boolean! @hasScope(scope: "workspace:delete")
   update(input: WorkspaceUpdateInput!): Workspace! @hasScope(scope: "workspace:update")
-  """
-  TODO: `@hasWorkspaceRole(role: WORKSPACE_ADMIN)` for role changes
-  """
   updateRole(input: WorkspaceRoleUpdateInput!): Workspace!
     @hasScope(scope: "workspace:update")
-  deleteRole(input: WorkspaceRoleDeleteInput!): Workspace!
-    @hasScope(scope: "workspace:update")
+    @hasWorkspaceRole(role: ADMIN)
```

Comment:
This ain't gonna work, IIRC @hasWorkspaceRole only works when attached to fields directly on the `Workspace` type. You can check the implementation - it relies on resolving what the workspace that you're referring to is based on `parent.id`

Reply:
ty for flagging I didn't realize that's how it worked. Using `authorizeResolver` now. In a previous PR, we opted to do this in the service function level, but here that go real messy real quick wrt testing and DI. Opted to do it in the resolver here (there's only one for roles now so it feels a bit better)